dbsnap_verify/__init__.py

```diff
@@ -0,0 +1,176 @@
+from rds_funcs import (
+    get_latest_snapshot,
+    get_database_description,
+    restore_from_latest_snapshot,
+    modify_db_instance_for_verify,
+    dbsnap_verify_db_id,
+    destroy_database,
+    destroy_database_subnet_group,
+    rds_event_messages,
+)
+
+from state_doc import (
+    current_state,
+    transition_state,
+    get_or_create_state_doc,
+)
+
+from time_funcs import (
+    tomorrow_timestamp,
+    now_datetime,
+    timestamp_to_datetime,
+    datetime_to_date_str,
+    three_days_prior,
+)
+
+import boto3
+
+from sys import stdout
+import logging
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+logger.addHandler(logging.StreamHandler(stdout))
+
+
+def wait(state_doc, rds_session):
+    min_timestamp = state_doc["snapshot_minimum_timestamp"]
+    min_datetime = timestamp_to_datetime(min_timestamp)
+    logger.info("Looking for a snapshot of {} older then {}".format(
+        state_doc["database"],
+        datetime_to_date_str(min_datetime)
+    ))
+    snapshot_desc = get_latest_snapshot(rds_session, state_doc["database"])
+    if snapshot_desc["SnapshotCreateTime"] >= min_datetime:
+        # if the latest snapshot is older then the minimum, restore it.
+        restore(state_doc, rds_session)
+    elif min_datetime >= three_days_prior(now_datetime()):
```

Comment:
What's magical about three days here?

Reply:
Nothing magic its just a date in the past that we expect snapshots to be older then. Now that I'm thinking about it 3 days is sort of Janky, I think maybe just getting this to 0 might make the most sense. That way it triggers on this first execution.